<?xml version="1.0" ?>
<launch>

  <arg name="driver_ns"		default="dynamixel_driver"/>
  <arg name="usb_port"		default="/dev/ttyUSB0"/>
  <arg name="dxl_baud_rate"	default="1000000"/>
  <arg name="control_period"	default="0.010"/>
  <arg name="dynamixel_info"	default="dynamixel_info"/>
  <arg name="tool_names"	default="screw_tool_m3,screw_tool_m4"/>
  <arg name="motor_ids"		default="2,3"/>
  <arg name="tool_number"	default="0"/>

  <group unless="$(eval tool_number == len(tool_names.split(',')))">
    <arg name="tool_name"
	 value="$(eval tool_names.split(',')[tool_number].strip())"/>
    <arg name="motor_id"
	 value="$(eval motor_ids.split(',')[tool_number].strip())"/>

    <node name="$(arg tool_name)_controller"
  	  pkg="aist_fastening_tools" type="screw_tool_controller"
  	  output="screen">
      <param name="driver_ns"		value="/$(arg driver_ns)"/>
      <param name="motor_id"		value="$(arg motor_id)"/>
      <param name="control_period"	value="$(arg control_period)"/>
    </node>

    <include file="$(dirname)/screw_tool_controllers.launch">
      <arg name="driver_ns"		value="$(arg driver_ns)"/>
      <arg name="usb_port"		value="$(arg usb_port)"/>
      <arg name="dxl_baud_rate"		value="$(arg dxl_baud_rate)"/>
      <arg name="control_period"	value="$(arg control_period)"/>
      <arg name="dynamixel_info"	value="$(arg dynamixel_info)"/>
      <arg name="tool_names"		value="$(arg tool_names)"/>
      <arg name="motor_ids"		value="$(arg motor_ids)"/>
      <arg name="tool_number"		value="$(eval tool_number + 1)"/>
    </include>
  </group>

  <group if ="$(eval tool_number == len(tool_names.split(',')))">
    <param name="dynamixel_info"	value="$(dirname)/../config/$(arg
					       dynamixel_info).yaml"/>

    <node name="$(arg driver_ns)"
	  pkg="dynamixel_workbench_controllers"
	  type="dynamixel_workbench_controllers"
          output="screen" args="$(arg usb_port) $(arg dxl_baud_rate)">
      <param name="use_moveit"			value="false"/>
      <param name="use_joint_states_topic"	value="false"/>
      <param name="use_cmd_vel_topic"		value="false"/>
      <rosparam subst_value="true">
        publish_period:   $(arg control_period)
        dxl_read_period:  $(arg control_period)
        dxl_write_period: $(arg control_period)
      </rosparam>
    </node>
  </group>

</launch>
